### PR TITLE
Add load test

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2315

Test result for local:

**Jmeter command**
```
jmeter -n -t http-post-request.jmx -l original.jtl -Jusers=10 -Jduration=300 -Jhost=192.168.64.3 -Jport=30966 -Jprotocol=http -Jpath=serv -Jresponse_size=50 -Jpayload="50B.json"
```
**Result**
```
summary +    142 in 00:00:05 =   29.0/s Avg:    33 Min:     9 Max:   125 Err:     0 (0.00%) Active: 1 Started: 1 Finished: 0
summary +   1916 in 00:00:30 =   63.9/s Avg:    59 Min:     8 Max:   258 Err:     0 (0.00%) Active: 6 Started: 6 Finished: 0
summary =   2058 in 00:00:35 =   59.0/s Avg:    57 Min:     8 Max:   258 Err:     0 (0.00%)
summary +   2988 in 00:00:30 =   99.6/s Avg:    86 Min:     5 Max:   287 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =   5046 in 00:01:05 =   77.8/s Avg:    74 Min:     5 Max:   287 Err:     0 (0.00%)
summary +   3217 in 00:00:30 =  107.3/s Avg:    93 Min:    12 Max:   255 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =   8263 in 00:01:35 =   87.1/s Avg:    81 Min:     5 Max:   287 Err:     0 (0.00%)
summary +   3594 in 00:00:30 =  119.6/s Avg:    83 Min:    10 Max:   283 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  11857 in 00:02:05 =   94.9/s Avg:    82 Min:     5 Max:   287 Err:     0 (0.00%)
summary +   3318 in 00:00:30 =  110.8/s Avg:    90 Min:    10 Max:   373 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  15175 in 00:02:35 =   98.0/s Avg:    84 Min:     5 Max:   373 Err:     0 (0.00%)
summary +   4529 in 00:00:30 =  150.7/s Avg:    66 Min:    10 Max:   298 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  19704 in 00:03:05 =  106.5/s Avg:    80 Min:     5 Max:   373 Err:     0 (0.00%)
summary +   5394 in 00:00:30 =  179.8/s Avg:    55 Min:    11 Max:   188 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  25098 in 00:03:35 =  116.8/s Avg:    74 Min:     5 Max:   373 Err:     0 (0.00%)
summary +   5412 in 00:00:30 =  180.8/s Avg:    55 Min:     9 Max:   195 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  30510 in 00:04:05 =  124.6/s Avg:    71 Min:     5 Max:   373 Err:     0 (0.00%)
summary +   5806 in 00:00:30 =  193.6/s Avg:    51 Min:     8 Max:   296 Err:     0 (0.00%) Active: 10 Started: 10 Finished: 0
summary =  36316 in 00:04:35 =  132.1/s Avg:    68 Min:     5 Max:   373 Err:     0 (0.00%)
summary +   4932 in 00:00:25 =  195.9/s Avg:    51 Min:     7 Max:   180 Err:     0 (0.00%) Active: 0 Started: 10 Finished: 10
summary =  41248 in 00:05:00 =  137.5/s Avg:    66 Min:     5 Max:   373 Err:     0 (0.00%)

```
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: load-tests/order_management_service/deployment/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+spec:
+  rules:
+    - host: bal.perf.test
+      http:
+        paths:
+          - path: "/"
+            pathType: Prefix
+            backend:
+              service:
+                name: order-managemen
```
Comment (daneshk) [inline load-tests/order_management_service/deployment/ingress.yaml:19]:
```suggestion
                name: order-management
```

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam how do we get the jwt token and how it is validated

**Comment by kalaiyarasiganeshalingam** [comment]
> @kalaiyarasiganeshalingam how do we get the jwt token and how it is validated
I have created a token by using [this sample](https://ballerina.io/learn/by-example/security-jwt-issue-validate.html?is_ref_by_example=true), passed that token through the [request](https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-ae3a8c5f950c17f79a41a14634868cc52dd9fe25d734aa7d4e285f99f80c3c45R65), and validated it [here](https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-3da2a97fc844ad4ea6bcc6c98ed881e7aa0a896b8979a0d013456a94d9a7768dR50).

File: load-tests/order_management_service/deployment/ingress.yaml (hunk 1 above)
Comment (ldclakmal) [reply load-tests/order_management_service/deployment/ingress.yaml:17]: @daneshk we have to use the auto generated name in the deployment yaml and it has a max length of 15 chars. that's why the name appeared as so. I also agree that is weird. @dilanSachi do you know the reason for this length?

Comment (dilanSachi) [reply load-tests/order_management_service/deployment/ingress.yaml:17]: Im not sure either. looping in @xlight05

**Comment by ldclakmal** [comment]
> @kalaiyarasiganeshalingam how do we get the jwt token and how it is validated
> 
> Updated now.
> 
> Token creation: https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-24d936df4e48fa103c081da20d3b66c83d1cb7a969307c7c284e182b2035cf0cR19
> 
> Validate:https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-3da2a97fc844ad4ea6bcc6c98ed881e7aa0a896b8979a0d013456a94d9a7768dR50

Just to clarify, the token generation happens for each and every request according to the provider issuer configurations. Therefore the generated JWT is unique to each request. Token validation also happens for each and every request and it also uses a cache to store the validated JWTs. But since, the token is unique to each request, the cache should behave as configured. All the incoming tokens are checked against cache (obviously it is not there) and store it based on the capacity and eviction policy. IMO, may be we can add another load-test to with the `cleanupInterval` as well, in order to observe the behaviour of 2 evictions at the same time. Just a suggestion.

**Review by ldclakmal — APPROVED** [review]
LGTM

**Comment by daneshk** [comment]
> > @kalaiyarasiganeshalingam how do we get the jwt token and how it is validated
> > Updated now.
> > Token creation: https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-24d936df4e48fa103c081da20d3b66c83d1cb7a969307c7c284e182b2035cf0cR19
> > Validate:https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-3da2a97fc844ad4ea6bcc6c98ed881e7aa0a896b8979a0d013456a94d9a7768dR50
> 
> Just to clarify, the token generation happens for each and every request according to the provider issuer configurations. Therefore the generated JWT is unique to each request. Token validation also happens for each and every request and it also uses a cache to store the validated JWTs. But since, the token is unique to each request, the cache should behave as configured. All the incoming tokens are checked against cache (obviously it is not there) and store it based on the capacity and eviction policy. IMO, may be we can add another load-test to with the `cleanupInterval` as well, in order to observe the behaviour of 2 evictions at the same time. Just a suggestion.

+1. yes, good suggestion

File: load-tests/order_management_service/deployment/ingress.yaml (hunk 1 above)
Comment (daneshk) [reply load-tests/order_management_service/deployment/ingress.yaml:17]: ack

**Review by daneshk — APPROVED** [review]

**Comment by kalaiyarasiganeshalingam** [comment]
> > > @kalaiyarasiganeshalingam how do we get the jwt token and how it is validated
> > > Updated now.
> > > Token creation: https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-24d936df4e48fa103c081da20d3b66c83d1cb7a969307c7c284e182b2035cf0cR19
> > > Validate:https://github.com/ballerina-platform/module-ballerina-cache/pull/355/files#diff-3da2a97fc844ad4ea6bcc6c98ed881e7aa0a896b8979a0d013456a94d9a7768dR50
> > 
> > 
> > Just to clarify, the token generation happens for each and every request according to the provider issuer configurations. Therefore the generated JWT is unique to each request. Token validation also happens for each and every request and it also uses a cache to store the validated JWTs. But since, the token is unique to each request, the cache should behave as configured. All the incoming tokens are checked against cache (obviously it is not there) and store it based on the capacity and eviction policy. IMO, may be we can add another load-test to with the `cleanupInterval` as well, in order to observe the behaviour of 2 evictions at the same time. Just a suggestion.
> 
> +1. yes, good suggestion

I will add this. Please find the issue to track this:
https://github.com/ballerina-platform/ballerina-standard-library/issues/2461

**Pull request merged**